packages/kbn-dev-utils/src/run/metrics.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import path from 'path';
+import { REPO_ROOT } from '@kbn/utils';
+import { CiStatsReporter } from '../ci_stats_reporter';
+import { ToolingLog } from '../tooling_log';
+
+export type MetricsMeta = Map<string, string | boolean | number>;
+
+export class Metrics {
+  private reporter: CiStatsReporter;
+  meta: MetricsMeta = new Map();
+  startTime: number;
+  filePath: string;
+
+  constructor(log: ToolingLog) {
+    this.reporter = CiStatsReporter.fromEnv(log);
+    this.meta = new Map();
+    this.startTime = Date.now();
+    this.filePath = path.relative(REPO_ROOT, process.argv[1]).replace('.js', '');
```

Comment:
@spalger `caller-callsite` didn't work. But I think this solution should do what we need without the added dependency anyways.

Reply:
Is the second argument resolved to an absolute path relative to `process.cwd()` before the relative path is calculated? I feel like we should do that explicitly.

Reply:
The second argument is an absolute path.

Reply:
![image](https://user-images.githubusercontent.com/40265/135169837-44581bea-9739-4135-bb35-ec5b4272251e.png)

Reply:
Oh, process.argv[1] is resolved by node? TIL

Reply:
Yup yup